pkg/controller/operator.go

```diff
@@ -112,6 +113,23 @@ func (s *Support) Run(controller *controllercmd.ControllerContext) error {
 	authorizer := clusterauthorizer.New(configObserver)
 	insightsClient := insightsclient.New(nil, 0, "default", authorizer, configPeriodic)
 
+	// read proxy CA bundle
+	insightsClient.GetTrustedCABundleFunc = func() (*x509.CertPool, error) {
```

Comment:
Given now the function doesn't depend on anything from this context, would be simpler just put it directly to `insightsclient.go` or you did that more for testing purposes?

Reply:
Even for testing purposes, it would be good to set the default function in the `insightsclient.New` function, otherwise, there is a possibility for `GetTrustedCABundleFunc` to return `nil`, risking `panic: runtime error: invalid memory address or nil pointer dereference` eventually. I know the chances are low, but if we can choose between two approaches, let's go the safer one.